galaxy_utils/sequence/fastq.py

```diff
@@ -535,7 +539,20 @@ def _get_med_pos( size ):
 
 
 class fastqReader( Iterator ):
-    def __init__( self, fh, format='sanger', apply_galaxy_conventions=False ):
+    def __init__( self, fh=None, format='sanger', apply_galaxy_conventions=False, path=None ):
+        if fh is None:
```

Comment:
minor detail, why not reduce the copy'n paste by making a new `get_filehandle` function with something like `self.fh = get_filehandle(fh, path))` ?